src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,145 @@
+                book.setId(id);
+                book.setTitle(title);
+                book.setPrice(price);
+                return Optional.of(book);
+            }
+
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't find the book: " + e);
+        }
+        return Optional.empty();
+    }
+
+
+    @Override
+    public List<Book> findAll() {
+        String query = "SELECT * FROM book";
+        List<Book> books = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS);
+
+            ResultSet resultSet = statement.executeQuery();
+
+            while (resultSet.next()) {
+                Book book = new Book();
+                Long id = resultSet.getObject(1, Long.class);
+                String title = resultSet.getString(2);
+                BigDecimal price = resultSet.getBigDecimal(3);book.setId(id);
+                book.setTitle(title);
+                book.setPrice(price);
+                books.add(book);
+            }
+            return books;
+
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't add new book " + e);
+        }
+    }
+
+    @Override
+    public Book update(Book book) {
+        String query = "UPDATE book " +
+                "SET title = ?, price = ? " +
+                "WHERE id = ? ;";
+        try (Connection connection = ConnectionUtil.getConnection()){
+            PreparedStatement statement = connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS);
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            statement.setLong(3, book.getId());
+            int affectedRows = statement.executeUpdate();
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                book.setId(id);
+            }
+
+
+            if (affectedRows < 1) {
+                throw new RuntimeException("Expected to update at least 1 row, but inserted 0 rows");
+            }
+
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't add new book " + e);
+        }
+        return book;
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        String query = "DELETE FROM BOOK WHERE id = ? ;";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS);
+            statement.setLong(1, id);
+
+            int affectedRows = statement.executeUpdate();
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+
+            if (generatedKeys.next()) {
+                return true;
+            }
```

Comment:
In the `deleteById` method, checking for generated keys is incorrect since a DELETE operation does not generate keys. Instead, check if `affectedRows` is greater than 0 to determine if the deletion was successful.